test/patterns.py

```diff
@@ -0,0 +1,66 @@
+#
+# Copyright 2018-2018 Ghent University
+#
+# This file is part of vsc-base,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# the Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/hpcugent/vsc-base
+#
+# vsc-base is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Library General Public License as
+# published by the Free Software Foundation, either version 2 of
+# the License, or (at your option) any later version.
+#
+# vsc-base is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Library General Public License for more details.
+#
+# You should have received a copy of the GNU Library General Public License
+# along with vsc-base. If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Unit tests for the patterns.
+"""
+from vsc.install.testing import TestCase
+
+from vsc.utils.patterns import Singleton
+
+
+class ABC(object):
+    __metaclass__ = Singleton
+
+    def __init__(self, value):
+        """
+        Singleton metaclass with init that has args is probably the most stupid thing to have,
```

Comment:
I do not grok this comment. Why not be able to set stuff?

Reply:
check the tests, tell me that it makes sense

Reply:
@itkovian if 2 things need the singleton to be initialised in a different way, it can't and shouldn't be a singleton?